WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductFromImage/AddProductFromImageView.swift

```diff
@@ -0,0 +1,85 @@
+import SwiftUI
+import Yosemite
+
+/// Product data from the "add product from image" form.
+struct AddProductFromImageData {
+    let name: String
+    let description: String
+    // TODO: 10180 - image from media picker
+}
+
+/// Hosting controller for `AddProductFromImageView`.
+final class AddProductFromImageHostingController: UIHostingController<AddProductFromImageView> {
+    init(siteID: Int64,
+         completion: @escaping (AddProductFromImageData) -> Void) {
+        super.init(rootView: AddProductFromImageView(siteID: siteID, completion: completion))
+    }
+
+    required dynamic init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+}
+
+/// A form to create a product from an image, where any texts in the image can be scanned to generate product details with Jetpack AI.
+struct AddProductFromImageView: View {
+    private let completion: (AddProductFromImageData) -> Void
+    @StateObject private var viewModel: AddProductFromImageViewModel
+
+    init(siteID: Int64,
+         stores: StoresManager = ServiceLocator.stores,
+         completion: @escaping (AddProductFromImageData) -> Void) {
+        self.completion = completion
+        self._viewModel = .init(wrappedValue: AddProductFromImageViewModel())
+    }
+
+    var body: some View {
+        Form {
+            Section {
+                // TODO: 10180 - use `TextEditor` with a placeholder overlay
+                TextField(Localization.nameFieldPlaceholder, text: $viewModel.name)
+                    .lineLimit(2)
+                    .fixedSize(horizontal: false, vertical: true)
+                TextField(Localization.descriptionFieldPlaceholder, text: $viewModel.description)
+                    .lineLimit(5)
+                    .fixedSize(horizontal: false, vertical: true)
+            }
+        }
+        .navigationTitle(Localization.title)
+        .toolbar {
+            ToolbarItem(placement: .navigationBarTrailing) {
```

Comment:
💡 We can also use `.confirmationAction` here to get the most suitable position for the button based on its feature.

Reply:
Nice, TIL. Updated in https://github.com/woocommerce/woocommerce-ios/pull/10184/commits/b7bc1210dfc280f342dd8fd529323c14b748693a